File: xdr_temp_size.go
package main

import (
	"fmt"
	"io/ioutil"
	"log"
	"os"
	"path/filepath"
	"strings"
	"time"

	"github.com/karrick/godirwalk"
)

func main() {
	// Define the folder to scan
	folderToScan := "C:\\Users"

	// Define the output file path
	outputFolder := "C:\\Temp\\CortexXDR"
	if _, err := os.Stat(outputFolder); os.IsNotExist(err) {
		os.Mkdir(outputFolder, 0755)
	}
	hostname, err := os.Hostname()
	if err != nil {
		log.Fatal(err)
	}
	outputFile := filepath.Join(outputFolder, fmt.Sprintf("%s_output.txt", hostname))

	// Define the headers for the output file
	headers := "Path|Filename|File Creation|File Size"

	// Initialize the total size variable
	totalSize := int64(0)

	// Walk through each file and folder
	err = godirwalk.Walk(folderToScan, &godirwalk.Options{
		Callback: func(osPathname string, de *godirwalk.Dirent) error {
			if !de.IsRegular() {
				return nil
			}

			// Get the file path
			filePath := filepath.Dir(osPathname)

			// Get the file name
			fileName := filepath.Base(osPathname)

			// Get the file creation time
			fileCreationTime := de.ModTime().Format("2006-01-02 03:04:05 PM")

			// Get the file size in bytes
			fileSize := de.Size()

			// Convert the file size to a human-readable format
			sizeSuffixes := []string{"B", "KB", "MB", "GB", "TB"}
			sizeSuffixIndex := 0
			for fileSize > 1024 && sizeSuffixIndex < len(sizeSuffixes)-1 {
				fileSize /= 1024
				sizeSuffixIndex++
			}
			fileSizeString := fmt.Sprintf("%d%s", fileSize, sizeSuffixes[sizeSuffixIndex])

			// Update the total size
			totalSize += de.Size()

			// Write the file info to the output file
			fileInfo := fmt.Sprintf("%s|%s|%s|%s\n", filePath, fileName, fileCreationTime, fileSizeString)
			err := ioutil.WriteFile(outputFile, []byte(fileInfo), 0644)
			if err != nil {
				log.Fatalf("Error writing file info to output file: %v\n", err)
			}

			return nil
		},
		ErrorCallback: func(osPathname string, err error) godirwalk.ErrorAction {
			log.Printf("Error scanning %q: %v\n", osPathname, err)
			return godirwalk.SkipNode
		},
		Unsorted: true, // Traverse the directory in an arbitrary order
	})
	if err != nil {
		log.Fatalf("Error scanning directory: %v\n", err)
	}

	// Convert the total size to a human-readable format
	sizeSuffixes := []string{"B", "KB", "MB", "GB", "TB"}
	sizeSuffixIndex := 0
	for totalSize > 1024 && sizeSuffixIndex < len(sizeSuffixes)-1 {
		totalSize /= 1024
		sizeSuffixIndex++
	}
	totalSizeString := fmt.Sprintf("%d%s", totalSize, sizeSuffixes[sizeSuffixIndex])

	// Write the total size to the output file

File: xdr_temp_size.py
import os
import socket
import time
import subprocess

# Define the folder to scan
folder_to_scan = ["C:\\Windows\\Temp", "C:\\Temp"]

# Define the output file path
output_folder = "C:\\Temp\\CortexXDR"
if not os.path.exists(output_folder):
    os.makedirs(output_folder)
output_file = os.path.join(output_folder, f"{socket.gethostname()}_output.txt")

# Define the headers for the output file
headers = ["Path", "Filename", "File Creation", "File Size"]

# Open the output file
with open(output_file, "w") as f:
    # Write the headers to the file
    f.write("|".join(headers) + "\n")

    # Initialize the total size variable
    total_size = 0

    # Loop through each folder and subfolder
    for folder in folder_to_scan:
        for dirpath, dirnames, filenames in os.walk(folder):
            for filename in filenames:
                # Get the full path to the file
                file_path = os.path.join(dirpath, filename)

                # Get the file creation time
                file_creation_time = os.path.getctime(file_path)
                file_creation_time = time.strftime("%Y-%m-%d %I:%M:%S %p", time.localtime(file_creation_time))

                # Get the file size in bytes
                file_size = os.path.getsize(file_path)

                # Convert the file size to a human-readable format
                size_suffixes = ["B", "KB", "MB", "GB", "TB"]
                size_suffix_index = 0
                while file_size > 1024 and size_suffix_index < len(size_suffixes) - 1:
                    file_size /= 1024.0
                    size_suffix_index += 1
                file_size = "{:.2f} {}".format(file_size, size_suffixes[size_suffix_index])

                # Add the file size to the total size
                total_size += os.path.getsize(file_path)

                # Write the file details to the output file
                f.write("|".join([dirpath, filename, file_creation_time, file_size]) + "\n")

    # Convert the total size to a human-readable format
    size_suffix_index = 0
    while total_size > 1024 and size_suffix_index < len(size_suffixes) - 1:
        total_size /= 1024.0
        size_suffix_index += 1
    total_size = "{:.2f} {}".format(total_size, size_suffixes[size_suffix_index])

    # Write the total size to the output file
    f.write(f"\nTotal Size: {total_size}")

# Close the console window
subprocess.call('cmd /c exit /b', shell=True)
